storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java

```diff
@@ -162,6 +185,191 @@ public void testGetUserId() throws Exception {
+    private synchronized long getPidOfPosixProcess(Process p, List<String> errors) {
+        Class<? extends Process> pClass = p.getClass();
+        String pObjStr = ToStringBuilder.reflectionToString(p, ToStringStyle.SHORT_PREFIX_STYLE);
+        String pclassName = pClass.getName();
+        try {
+            if (pclassName.equals("java.lang.UNIXProcess")) {
+                Field f = pClass.getDeclaredField("pid");
+                f.setAccessible(true);
+                long pid = f.getLong(p);
+                f.setAccessible(false);
+                if (pid < 0) {
+                    errors.add("\t \"pid\" attribute in Process class " + pclassName + " returned -1, process=" + pObjStr);
+                }
+                return pid;
+            }
+            for (Field f : pClass.getDeclaredFields()) {
+                if (!f.getName().equalsIgnoreCase("pid")) {
+                    continue;
+                }
+                LOG.info("ServerUtilsTest.getPidOfPosixProcess(): found attribute {}#{}", pclassName, f.getName());
+                f.setAccessible(true);
+                long pid = f.getLong(p);
+                f.setAccessible(false);
+                if (pid < 0) {
+                    errors.add("\t \"pid\" attribute in Process class " + pclassName + " returned -1, process=" + pObjStr);
+                }
+                return pid;
+            }
+            // post JDK 9 there should be getPid() - future JDK-11 compatibility only for the sake of Travis test in community
+            try {
+                Method m = pClass.getDeclaredMethod("getPid");
+                LOG.info("ServerUtilsTest.getPidOfPosixProcess(): found method {}#getPid()\n", pclassName);
+                long pid = (Long)m.invoke(p);
+                if (pid < 0) {
+                    errors.add("\t \"getPid()\" method in Process class " + pclassName + " returned -1, process=" + pObjStr);
+                }
+                return pid;
+            } catch (SecurityException e) {
+                errors.add("\t getPid() method in Process class " + pclassName + " cannot be called: " + e.getMessage() + ", process=" + pObjStr);
+                return -1;
+            } catch (NoSuchMethodException e) {
+                // ignore and try something else
+            }
+            errors.add("\t Process class " + pclassName + " missing field \"pid\" and missing method \"getPid()\", process=" + pObjStr);
+            return -1;
+        } catch (Exception e) {
+            errors.add("\t Exception in Process class " + pclassName + ": " + e.getMessage() + ", process=" + pObjStr);
+            e.printStackTrace();
+            return -1;
+        }
+    }
+
+    /**
+     * Sleep for specified milliseconds and return true if sleep was interrupted.
+     *
+     * @param milliSeconds number of milliseconds to sleep
+     * @return true if sleep was interrupted, false otherwise.
+     */
+    private boolean sleepInterrupted(long milliSeconds) {
```

Comment:
this method already exist?

Reply:
I am not aware of any existing method with similar functionality.

Reply:
https://github.com/apache/storm/blob/master/storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java#L324
This method already exists in ServerUtilsTest (from your another merged PR)